indra/llui/lltabcontainer.cpp

```diff
@@ -2173,32 +2173,23 @@ S32 LLTabContainer::getTotalTabWidth() const
 
 void LLTabContainer::setTabVisibility( LLPanel const *aPanel, bool aVisible )
 {
-    for( tuple_list_t::const_iterator itr = mTabList.begin(); itr != mTabList.end(); ++itr )
-    {
-        LLTabTuple const *pTT = *itr;
-        if( pTT->mTabPanel == aPanel )
-        {
-            pTT->mVisible = aVisible;
-            break;
-        }
-    }
-
-    bool foundTab( false );
-    for( tuple_list_t::const_iterator itr = mTabList.begin(); itr != mTabList.end(); ++itr )
+    S32 num_tabs = S32(mTabList.size());
+    for (S32 i = 0; i < num_tabs; ++i)
     {
-        LLTabTuple const *pTT = *itr;
-        if( pTT->mVisible )
+        LLTabTuple* tuple = mTabList[i];
+        if( tuple->mTabPanel == aPanel )
         {
-            this->selectTab( itr - mTabList.begin() );
-            foundTab = true;
+            if (tuple->mVisible != aVisible)
+            {
+                tuple->mVisible = aVisible;
+                if (aVisible)
+                {
+                    this->selectTab(i);
+                    this->setVisible(true);
+                }
+                updateMaxScrollPos();
+            }
             break;
         }
     }
-
-    if( foundTab )
-        this->setVisible( true );
-    else
-        this->setVisible( false );
-
-    updateMaxScrollPos();
```

Comment:
LLTabContainer::setTabVisibility() was walking mTabList twice when it only needed to walk it once, so I cleaned up its implementation.  I added one slight change in behavior where it first checks to see if the visibility is actually changing before applying the setting.  Note: this subtly avoids some side-effects of the `this->selectTab(); this->setVisible(true);` where UI which hasn't actually changed can nevertheless become selected/focused.  I don't expect this to cause any UX problems.